katas/content/superdense_coding/Common.qs

```diff
@@ -0,0 +1,79 @@
+namespace Kata.Verification {
+    operation CreateEntangledPairWrapper_Reference(qs : Qubit[]) : Unit is Adj + Ctl {
+        let (qAlice, qBob) = (qs[0], qs[1]);
+        H(qAlice);
+        CNOT(qAlice, qBob);
+    }
+
+    operation EncodeMessageInQubit_Reference(qAlice : Qubit, message : (Bool, Bool)) : Unit {
+        let (bit1, bit2) = message;
+
+        if bit1 {
+            Z(qAlice);
+        }
+
+        if bit2 {
+            X(qAlice);
+        }
+    }
+
+    operation DecodeMessageFromQubits_Reference(qAlice : Qubit, qBob : Qubit) : (Bool, Bool) {
+        CNOT(qAlice, qBob);
+        H(qAlice);
+        return (MResetZ(qAlice) == One, MResetZ(qBob) == One);
+    }
+
+
+    // ------------------------------------------------------
+    // Helper operation that runs superdense coding protocol using two building blocks
+    // specified as first two parameters.
+    operation ComposeProtocol(
+        encodeOp : ((Qubit, (Bool, Bool)) => Unit),
+        decodeOp : ((Qubit, Qubit) => (Bool, Bool)),
+        message : (Bool, Bool)
+    ) : (Bool, Bool) {
+
+        use (qAlice, qBob) = (Qubit(), Qubit());
+
+        CreateEntangledPairWrapper_Reference([qAlice, qBob]);
+
+        encodeOp(qAlice, message);
+
+        let (bit1, bit2) = decodeOp(qAlice, qBob);
+
+        // We get this error if dont call Reset
+        // [ERR_ASSERTION]: Exercise "superdense_coding__bob_decodes_message" has compilation or runtime errors
+        //  when using the placeholder as solution. Compilation and runtime errors:
+        //   Message = runtime error: Qubit1 released while not in |0⟩ state
```

Comment:
You don't have to explain the Reset. It's a known behavior that qubits have to be in 0 state before release, you'll notice that, for example, in state preparation tasks, if the state prepared is not what we want (and thus after adjoint is applied the state is not 0), we reset the state before returning false, without much emphasis really:
```suggestion
```

Reply:
I thought of adding here in case someone ends up getting confused of the difference from original harness. But probably we can omit that, since its us developers only who would be referring old katas for short time :-)